### PR TITLE
Add apiErrorLogger action delegate to api client

### PR DESCRIPTION
**Pull request by joaomsa**

### CODE REVIEW

File: Synapse.RestClient/SynapseBaseApiClient.cs [hunk 1]
```diff
@@ -57,7 +59,9 @@ protected T ParseResponse<T>(IRestResponse resp)
             if (resp.StatusCode != System.Net.HttpStatusCode.OK)
             {
                 var error = JsonConvert.DeserializeObject<SynapseApiErrorResponse>(resp.Content, this._jsonSettings);
-                throw new SynapseApiErrorException(error);
+                var ex = new SynapseApiErrorException(error);
+                this._apiErrorLogger?.Invoke(ex);
```
Comment (jdconley) [inline Synapse.RestClient/SynapseBaseApiClient.cs:63]: Should check for null here. It could be passed in as null through the constructor.

Comment (joaomsa) [inline Synapse.RestClient/SynapseBaseApiClient.cs:63]: It checks using the C#6 null propagation with ?.

Comment (jdconley) [inline Synapse.RestClient/SynapseBaseApiClient.cs:63]: Oh wow, totally skimmed over that.

**Comment by jdconley** [comment]
:shipit:

**Pull request merged**